sky/shell/platform/ios/semantics_listener_impl.mm

```diff
@@ -0,0 +1,245 @@
+  for (size_t i = 0; i < input.size(); i++) {
+    result[i] = input[i];
+  }
+  return result;
+}
+
+void SemanticsListenerImpl::AccessibilityNode::invalidateGlobalGeometry() {
+  if (geometryDirty) {
+    return;
+  }
+  geometryDirty = true;
+  for (AccessibilityNode* child : children) {
+    child->invalidateGlobalGeometry();
+  }
+}
+
+SkMatrix44& SemanticsListenerImpl::AccessibilityNode::getGlobalTransform() {
+  if (geometryDirty) {
+    if (parent == NULL) {
+      globalTransform = transform;
+    } else {
+      SkMatrix44 parentTransform = parent->getGlobalTransform();
+      if (transform.isIdentity()) {
+        globalTransform = parentTransform;
+      } else if (parentTransform.isIdentity()) {
+        globalTransform = transform;
+      } else {
+        globalTransform = transform * parentTransform;
+      }
+    }
+  }
+  return globalTransform;
+}
+
+CGRect SemanticsListenerImpl::AccessibilityNode::getGlobalRect() {
+  if (geometryDirty) {
+    // TODO(tvolkert): optimize identity transform
+    SkMatrix44 globalTransform = getGlobalTransform();
+    float point1[] = {left, top, 0, 1};
+    float point2[] = {left + width, top, 0, 1};
+    float point3[] = {left + width, top + height, 0, 1};
+    float point4[] = {left, top + height, 0, 1};
+    float* transformedPoint1 = transformPoint(globalTransform, point1);
+    float* transformedPoint2 = transformPoint(globalTransform, point2);
+    float* transformedPoint3 = transformPoint(globalTransform, point3);
+    float* transformedPoint4 = transformPoint(globalTransform, point4);
+    float minX = std::min({transformedPoint1[0], transformedPoint2[0],
+                           transformedPoint3[0], transformedPoint4[0]});
+    float minY = std::min({transformedPoint1[1], transformedPoint2[1],
+                           transformedPoint3[1], transformedPoint4[1]});
+    float maxX = std::max({transformedPoint1[0], transformedPoint2[0],
+                           transformedPoint3[0], transformedPoint4[0]});
+    float maxY = std::max({transformedPoint1[1], transformedPoint2[1],
+                           transformedPoint3[1], transformedPoint4[1]});
+    float width = maxX - minX;
+    float height = maxY - minY;
+    globalRect = CGRectMake(minX, minY, width, height);
+    delete transformedPoint1;
+    delete transformedPoint2;
+    delete transformedPoint3;
+    delete transformedPoint4;
+  }
+  return globalRect;
+}
+
+// TODO(tvolkert): call map2() with x, y, w, h all at once
+float* SemanticsListenerImpl::AccessibilityNode::transformPoint(
+    SkMatrix44& transform,
+    float point[]) {
+  float* transformed = (float*)malloc(4 * sizeof(float));
+  if (transform.isIdentity()) {
+    std::copy(point, point + 4, transformed);
+  } else {
+    transform.map2(point, 1, transformed);
+  }
+  return transformed;
+}
+
+void SemanticsListenerImpl::AccessibilityNode::populateAccessibleElements(
+    NSMutableArray* accessibleElements) {
+  if (width > 0 && height > 0) {
```

Comment:
Are width and height members?  The pattern we typically use is m_width and m_height to show such.  At least that was in Blink, I think we might use width_ and height_ now.